src/components/carousel/__spec__.js

```diff
@@ -17,23 +17,128 @@ describe('Carousel', () => {
   });
 
   describe('componentWillMount', () => {
+    describe('when slideIndex is passed', () => {
+      it('sets the intial slide to the prop', () => {
+        const wrapper = shallow(
+          <Carousel data-element='bar' data-role='baz' slideIndex={ 1 }>
+            <Slide />
+            <Slide />
+            <Slide />
+          </Carousel>
+        );
+
+        expect(wrapper.instance().state.selectedSlideIndex).toEqual(1);
+      });
+    });
+
     describe('when initialSlideIndex is passed', () => {
       it('sets the intial slide to the prop', () => {
-        instance = TestUtils.renderIntoDocument(
-          <Carousel initialSlideIndex={ 1 }>
+        const wrapper = shallow(
+          <Carousel data-element='bar' data-role='baz' initialSlideIndex={ 1 }>
             <Slide />
             <Slide />
             <Slide />
           </Carousel>
         );
 
-        expect(instance.state.selectedSlideIndex).toEqual(1);
+        expect(wrapper.instance().state.selectedSlideIndex).toEqual(1);
       });
     });
 
     describe('when initialSelectedId is not passed', () => {
       it('defaults the initial slide to slide 0', () => {
-        expect(instance.state.selectedSlideIndex).toEqual(0);
+        const wrapper = shallow(
+          <Carousel data-element='bar' data-role='baz'>
+            <Slide />
+            <Slide />
+            <Slide />
+          </Carousel>
+        );
+
+        expect(wrapper.instance().state.selectedSlideIndex).toEqual(0);
+      });
+    });
+  });
+
+  describe('componentWillReceiveProps', () => {
+    const enableButtonsAfterTimeoutSpy = jasmine.createSpy(),
+        wrapper = shallow(
+          <Carousel data-element='bar' data-role='baz' initialSlideIndex={ 0 }>
+            <Slide />
+            <Slide />
+            <Slide />
+          </Carousel>
+    );
+
+    beforeEach(() => {
+      wrapper.instance().enableButtonsAfterTimeout = enableButtonsAfterTimeoutSpy;
+    });
+
+    it('navigate between slides correctly when the slideIndex prop changes', () => {
+      // Initial state
+      expect(wrapper.state().selectedSlideIndex).toEqual(0);
+      expect(wrapper.state().disabled).toBeFalsy();
+      expect(enableButtonsAfterTimeoutSpy).not.toHaveBeenCalled();
+
+      // Move to slide 2
+      wrapper.setProps({ slideIndex: 2 });
+      wrapper.rerender();
```

Comment:
`setProps` should rerender on it's own.

https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/setProps.md